File: BasicInputOutput/C/SwapNumbersBitwise.c
#include <stdio.h>

int main() {
    int a, b;
    printf("Enter Two Values: ");
    scanf("%d %d", &a, &b);

    printf("Before Swap: %d, %d\n", a, b);

    // Using Bitwise XOR to swap without a third variable
    a = a ^ b;
    b = a ^ b;
    a = a ^ b;

    printf("After Swap: %d, %d\n", a, b);

    return 0;
}

File: BasicInputOutput/Java/SwapNumbersBitwise.java

import java.util.Scanner;

class  SwapNumbersBitwise{
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        System.out.println("Enter Two Value : ");
        int a = sc.nextInt();
        int b = sc.nextInt();

        System.out.println("Before Swap : "+a+", "+b);

        a = a^b;
        b = a^b;
        a = a^b;

        System.out.println("After Swap : "+a+", "+b);
    }
}